File: codes/ruby/polymorphism/overriding.rb
#!/usr/bin/ruby -w

class Person
    def talk()
        puts "I am a person"
    end
end

class Student < Person
    def talk()
        puts "I am a student"
    end
end

pe = Person.new
st = Student.new
pe.talk # I am a person
st.talk # I am a student

File: codes/python/polymorphism/overriding.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

class Person(object):
    def talk(self):
        print("I am a person")

class Student(Person):
    def talk(self):
        print("I am a student")

pe = Person()
st = Student()
pe.talk() # I am a person
st.talk() # I am a student
